xbmc/platform/darwin/osx/network/NetworkOsx.cpp

```diff
@@ -180,27 +181,45 @@ std::vector<std::string> CNetworkOsx::GetNameServers()
 {
   std::vector<std::string> result;
 
-  FILE* pipe = popen("scutil --dns | grep \"nameserver\" | tail -n2", "r");
-  usleep(100000);
+  FILE* pipe = popen("scutil --dns | grep \"nameserver\" | uniq | tail -n2", "r");
+
   if (pipe)
   {
-    std::vector<std::string> tmpStr;
-    char buffer[256] = {'\0'};
-    if (fread(buffer, sizeof(char), sizeof(buffer), pipe) > 0 && !ferror(pipe))
+    int fileno = ::fileno(pipe);
+    fd_set read_fds;
+    FD_ZERO(&read_fds);
+    FD_SET(fileno, &read_fds);
+
+    struct timeval timeout;
+    timeout.tv_sec = 1; // Set the timeout to 1 second
+    timeout.tv_usec = 0;
+
+    int ready = select(fileno + 1, &read_fds, nullptr, nullptr, &timeout);
```

Comment:
afaik `poll` is preferred over `select`, also doesn't require using those FD_ macros

Reply:
Ok - moved to poll. I agree code is now a bit cleaner.